src/test/java/org/verdictdb/coordinator/MetaDataStatementsTest.java

```diff
@@ -0,0 +1,388 @@
+import org.verdictdb.commons.DatabaseConnectionHelpers;
+import org.verdictdb.connection.CachedDbmsConnection;
+import org.verdictdb.connection.DbmsConnection;
+import org.verdictdb.connection.JdbcConnection;
+import org.verdictdb.exception.VerdictDBDbmsException;
+import org.verdictdb.exception.VerdictDBException;
+import org.verdictdb.jdbc41.VerdictConnection;
+import org.verdictdb.sqlsyntax.*;
+
+import java.io.IOException;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+@RunWith(Parameterized.class)
+public class MetaDataStatementsTest {
+
+  private static Map<String, Connection> connMap = new HashMap<>();
+
+  private static Map<String, String> schemaMap = new HashMap<>();
+
+  private static Map<String, SqlSyntax> syntaxMap = new HashMap<>();
+
+  private static final String MYSQL_HOST;
+
+  private String database;
+
+  private static final String[] targetDatabases = {"mysql", "impala", "redshift", "postgresql"};
+
+  public MetaDataStatementsTest(String database) {
+    this.database = database;
+  }
+
+  static {
+    String env = System.getenv("BUILD_ENV");
+    if (env != null && (env.equals("GitLab") || env.equals("DockerCompose"))) {
+      MYSQL_HOST = "mysql";
+    } else {
+      MYSQL_HOST = "localhost";
+    }
+  }
+
+  private static final String MYSQL_DATABASE =
+      "data_type_test" + RandomStringUtils.randomAlphanumeric(4).toLowerCase();
+
+  private static final String MYSQL_USER = "root";
+
+  private static final String MYSQL_PASSWORD = "";
+
+  private static final String IMPALA_HOST;
+
+  static {
+    IMPALA_HOST = System.getenv("VERDICTDB_TEST_IMPALA_HOST");
+  }
+
+  private static final String IMPALA_DATABASE =
+      "data_type_test" + RandomStringUtils.randomAlphanumeric(4).toLowerCase();
+
+  private static final String IMPALA_USER = "";
+
+  private static final String IMPALA_PASSWORD = "";
+
+  private static final String REDSHIFT_HOST;
+
+  private static final String REDSHIFT_DATABASE = "dev";
+
+  private static final String REDSHIFT_USER;
+
+  private static final String REDSHIFT_PASSWORD;
+
+  private static final String POSTGRES_HOST;
+
+  private static final String POSTGRES_DATABASE = "test";
+
+  private static final String POSTGRES_USER = "postgres";
+
+  private static final String POSTGRES_PASSWORD = "";
+
+  static {
+    String env = System.getenv("BUILD_ENV");
+    if (env != null && (env.equals("GitLab") || env.equals("DockerCompose"))) {
+      POSTGRES_HOST = "postgres";
+    } else {
+      POSTGRES_HOST = "localhost";
+    }
+  }
+
+  static {
+    REDSHIFT_HOST = System.getenv("VERDICTDB_TEST_REDSHIFT_ENDPOINT");
+    REDSHIFT_USER = System.getenv("VERDICTDB_TEST_REDSHIFT_USER");
+    REDSHIFT_PASSWORD = System.getenv("VERDICTDB_TEST_REDSHIFT_PASSWORD");
+  }
+
+  private static final String SCHEMA_NAME =
```

Comment:
Is this for Postgres? Please make it explicit. Note that postgres and redshift are the only systems that distinguish "database" from "schema".